sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -230,6 +238,111 @@ public void addEvent(String eventName, Map<String, Object> traceEventAttributes,
         }
     }
 
+    /**
+     * Returns a {@link SpanBuilder} to create and start a new child {@link Span} with parent
+     * being the designated {@link Span}.
+     *
+     * @param spanBuilder SpanBuilder for the span. Must be created before calling this method
+     * @param setAttributes Callback to populate attributes for the span.
+     * @param makeCurrent Flag indicating if span should be current after start.
+     *
+     * @return A {@link Context} with created {@link Span}.
+     */
+    private Context startSpanInternal(SpanBuilder spanBuilder,
+                                      java.util.function.BiConsumer<Span, Context> setAttributes,
+                                      boolean makeCurrent,
```

Comment:
I think this option makes it easy to leak thread local state, e.g. calling start() and end() from different threads

Reply:
not sure what's the concern here. This is a flag that tells to call `makeCurrent` on OTel Span or not.. `Context` is still used to match start/end calls.
For the current span propagation between threads we fully rely on OTel agent.

Reply:
it exists to enable the case when auto-instrumentation happens under the CosmosDB span and should be a child of it. Currently, it's a sibling because Cosmos span is not current.

Reply:
For Cosmos we need to consume new api  public Context start(String spanName, StartSpanOptions options, Context context)  which StartSpanOptions.setMakeCurrent(true) ? 
Also if the thread switch inside cosmos stack before making the http call , will http span will still be the child of cosmos ?